### PR TITLE
[paddle-pipelines] Add Chinese PDF preprocessing & text splitter

### PR DESCRIPTION
**Pull request by w5688414**
### Description

+ PDFToTextConverter

```
from pipelines.nodes import PDFToTextConverter, PreProcessor
from pathlib import Path

converter = PDFToTextConverter(
    remove_numeric_tables=True, merge_sentences=True)
doc = converter.convert(file_path=Path("./data/pdf_files/dev_demo.pdf"), meta=None)

processor = PreProcessor(
    clean_empty_lines=True,
    clean_whitespace=True,
    clean_header_footer=True,
    split_by="word",
    split_length=50,
    split_overlap=0
)
print(doc)
docs = processor.process(doc)
print(docs)
```

results:

```
[{'content': '兴证策略认为，最恐慌的时候已经过去，未来一个月市场迎来阶段性修复窗口。 从海外市场表现看，对俄乌冲突的恐慌情绪已显著释放，海外权益市场也从单边下跌转入双向波动。 长期，继续聚焦科技创新的五大方向。1)新能源(新能源汽车、光伏、风电、特高压等)，2)新一代信息通信技术(人工智能、大数据、云计算、5G 等)，3)高端制造(智能数控机床、机器人、先进轨交装备等)，4)生物医药(创新药、CXO、医疗器械和诊断设备等)，5)军工(导弹设备、军工电子元器件、空间站、航天飞机等)。  ', 'content_type': 'text', 'meta': None}]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 3634.58docs/s]
[{'content': '兴证策略认为，最恐慌的时候已经过去，未来一个月市场迎来阶段性修复窗口。 从海外市场表现看，对俄乌冲突', 'content_type': 'text', 'meta': {'_split_id': 0}}, {'content': '的恐慌情绪已显著释放，海外权益市场也从单边下跌转入双向波动。 长期，继续聚焦科技创新的五大方向。1)', 'content_type': 'text', 'meta': {'_split_id': 1}}, {'content': '新能源(新能源汽车、光伏、风电、特高压等)，2)新一代信息通信技术(人工智能、大数据、云计算、5G ', 'content_type': 'text', 'meta': {'_split_id': 2}}, {'content': '等)，3)高端制造(智能数控机床、机器人、先进轨交装备等)，4)生物医药(创新药、CXO、医疗器械和', 'content_type': 'text', 'meta': {'_split_id': 3}}, {'content': '诊断设备等)，5)军工(导弹设备、军工电子元器件、空间站、航天飞机等)。', 'content_type': 'text', 'meta': {'_split_id': 4}}]
```

+ text splitter

```
from pipelines.nodes import CharacterTextSplitter

text_splitter = CharacterTextSplitter(chunk_size=50, chunk_overlap=0)

TEXT = """
This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in
paragraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1.\f

This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in
paragraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2.

This is a sample sentence in paragraph_3. This is a sample sentence in paragraph_3. This is a sample sentence in
paragraph_3. This is a sample sentence in paragraph_3. This is to trick the test with using an abbreviation\f like Dr.
in the sentence.
"""


docs = text_splitter.split_text(TEXT)

print(docs)

from pipelines.pipelines import Pipeline
pipe = Pipeline()

pipe.add_node(component=text_splitter, name="Splitter", inputs=["Query"])

result = pipe.run(query=TEXT)

print(result)
print(len(result['documents']))
```

```
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
WARNING - pipelines.nodes.preprocessor.text_splitter -  Created a chunk of size 211, which is longer than the specified 50
WARNING - pipelines.nodes.preprocessor.text_splitter -  Created a chunk of size 209, which is longer than the specified 50
['This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in\nparagraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1.', 'This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in\nparagraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2.', 'This is a sample sentence in paragraph_3. This is a sample sentence in paragraph_3. This is a sample sentence in\nparagraph_3. This is a sample sentence in paragraph_3. This is to trick the test with using an abbreviation\x0c like Dr.\nin the sentence.']
WARNING - pipelines.nodes.preprocessor.text_splitter -  Created a chunk of size 211, which is longer than the specified 50
WARNING - pipelines.nodes.preprocessor.text_splitter -  Created a chunk of size 209, which is longer than the specified 50
{'documents': ['This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in\nparagraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1.', 'This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in\nparagraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2.', 'This is a sample sentence in paragraph_3. This is a sample sentence in paragraph_3. This is a sample sentence in\nparagraph_3. This is a sample sentence in paragraph_3. This is to trick the test with using an abbreviation\x0c like Dr.\nin the sentence.'], 'root_node': 'Query', 'params': {}, 'query': '\nThis is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in\nparagraph_1. This is a sample sentence in paragraph_1. This is a sample sentence in paragraph_1.\x0c\n\nThis is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in\nparagraph_2. This is a sample sentence in paragraph_2. This is a sample sentence in paragraph_2.\n\nThis is a sample sentence in paragraph_3. This is a sample sentence in paragraph_3. This is a sample sentence in\nparagraph_3. This is a sample sentence in paragraph_3. This is to trick the test with using an abbreviation\x0c like Dr.\nin the sentence.\n', 'node_id': 'Splitter'}
3
```

### CODE REVIEW

File: pipelines/pipelines/nodes/preprocessor/text_splitter.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (sijunhe) [inline pipelines/pipelines/nodes/preprocessor/text_splitter.py:2]: copyright加上langchain

File: pipelines/tests/nodes/preprocessor/test_text_splitter.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (sijunhe) [inline pipelines/tests/nodes/preprocessor/test_text_splitter.py:2]: copyright 加上langchain

File: pipelines/pipelines/nodes/preprocessor/text_splitter.py (hunk 1 above)
Comment (w5688414) [reply pipelines/pipelines/nodes/preprocessor/text_splitter.py:2]: 已添加

File: pipelines/tests/nodes/preprocessor/test_text_splitter.py (hunk 2 above)
Comment (w5688414) [reply pipelines/tests/nodes/preprocessor/test_text_splitter.py:2]: 已添加

**Review by sijunhe — APPROVED** [review]
lgtm

**Pull request merged**